### PR TITLE
Implement the missing Series.swaplevel

### PR DESCRIPTION
**Pull request by xinrong-databricks**

### CODE REVIEW

**Comment by ueshin** [comment]
FYI: You have to remove the API from "missing" files and add a link in the doc.

**Comment by itholic** [comment]
You can find the missing files in `databricks/koalas/missing/series.py` and doc link in `docs/source/reference/series.rst` :)

File: databricks/koalas/tests/test_series.py [hunk 1]
```diff
@@ -1824,6 +1824,35 @@ def test_squeeze(self):
...
 
+    def test_swaplevel(self):
+        # MultiIndex with two levels
+        arrays = [[1, 1, 2, 2], ["red", "blue", "red", "blue"]]
+        pidx = pd.MultiIndex.from_arrays(arrays, names=("number", "color"))
+        pser = pd.Series(["a", "b", "c", "d"], index=pidx)
+        kser = ks.from_pandas(pser)
+        self.assert_eq(pser.swaplevel(), kser.swaplevel())
+        self.assert_eq(pser.swaplevel(0, 1), kser.swaplevel(0, 1))
+        self.assert_eq(pser.swaplevel("number", "color"), kser.swaplevel("number", "color"))
```
Comment (ueshin) [inline databricks/koalas/tests/test_series.py:1835]: Just curious: what if we specify the same value for both arguments? `ser.swaplevel(1, 1)`.

File: databricks/koalas/tests/test_series.py [hunk 2]
```diff
@@ -1824,6 +1824,35 @@ def test_squeeze(self):
         kser = ks.from_pandas(pser)
         self.assert_eq(kser.squeeze(), pser.squeeze())
 
+    def test_swaplevel(self):
+        # MultiIndex with two levels
```
Comment (ueshin) [inline databricks/koalas/tests/test_series.py:1828]: What if the index is not multi index?

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -2440,6 +2440,76 @@ def sort_index(
         else:
             return first_series(kdf)
 
+    def swaplevel(self, i=-2, j=-1, copy=True) -> "Series":
```
Comment (ueshin) [inline databricks/koalas/series.py:2443]: Could you add a type annotation for `copy`?

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -2440,6 +2440,76 @@ def sort_index(
...
+
+        for index in (i, j):
+            if not isinstance(index, int) and index not in self.index.names:
+                raise KeyError("Level %s not found" % index)
+
+        i = i if isinstance(i, int) else self.index.names.index(i)
+        j = j if isinstance(j, int) else self.index.names.index(j)
+
+        for index in (i, j):
+            if index >= len(self.index.names) or index < -len(self.index.names):
```
Comment (ueshin) [inline databricks/koalas/series.py:2499]: `self._internal.index_level` instead of `len(self.index.names)` ?

File: databricks/koalas/series.py [hunk 5]
```diff
@@ -2440,6 +2440,76 @@ def sort_index(
...
+                raise IndexError(
+                    "Too many levels: Index of the series has only %s levels, "
+                    "%s is not a valid level number" % (len(self.index.names), index)
+                )
+
+        index_map = list(zip(self._internal.index_spark_column_names, self._internal.index_names))
+        index_map[i], index_map[j], = index_map[j], index_map[i]
+        index_spark_column_names, index_names = zip(*index_map)
+        internal = self._kdf._internal.copy(
```
Comment (ueshin) [inline databricks/koalas/series.py:2508]: `self._internal` instead of`self._kdf._internal` ?

File: databricks/koalas/tests/test_series.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_series.py:1835]: `IndexError: Too many levels: Index of the series has only 1 levels, -2 is not a valid level number`.

That's a good test case! Let me add it.

File: databricks/koalas/tests/test_series.py (hunk 2 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_series.py:1828]: `IndexError: Too many levels: Index of the series has only 1 levels, -2 is not a valid level number`.

That's a good test case! Let me add it.

File: databricks/koalas/series.py (hunk 3 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py:2443]: Sounds good, added!

File: databricks/koalas/tests/test_series.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_series.py:1835]: `self.assert_eq(kser, kser.swaplevel(1, 1))`

The series wouldn't change. Added test cases for `ser.swaplevel(1, 1)`.

File: databricks/koalas/series.py (hunk 4 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: Makes sense! Modified.

File: databricks/koalas/series.py (hunk 5 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py:2508]: May I ask what's their difference at usage?

I see `return self._kdf._internal.select_column(self._column_label)` is for `self._internal`.

Comment (ueshin) [reply databricks/koalas/series.py]: `self._kdf._internal` contains other columns from the anchor DataFrame.

E.g.,:

```py
>>> midx = pd.MultiIndex.from_arrays([['a', 'b'], [1, 2]], names = ['word', 'number'])
>>> kdf = ks.DataFrame({'a': [1, 2], 'b': ['x', 'y']}, index=midx)
>>> kdf
             a  b
word number
a    1       1  x
b    2       2  y
>>> kdf.b.swaplevel()
number  word
1       a       1
2       b       2
Name: a, dtype: int64
```

should be:

```py
>>> kdf.b.swaplevel()
number  word
1       a       x
2       b       y
Name: b, dtype: object
```

File: databricks/koalas/tests/test_series.py [hunk 6]
```diff
@@ -1824,6 +1824,38 @@ def test_squeeze(self):
...
+        self.assert_eq(pser.swaplevel(0, 2), kser.swaplevel(0, 2))
+        self.assert_eq(pser.swaplevel(1, 2), kser.swaplevel(1, 2))
+        self.assert_eq(pser.swaplevel(1, 1), kser.swaplevel(1, 1))
+        self.assert_eq(pser.swaplevel(-1, -2), kser.swaplevel(-1, -2))
+        self.assert_eq(pser.swaplevel("number", "color"), kser.swaplevel("number", "color"))
+        self.assert_eq(pser.swaplevel("number", "size"), kser.swaplevel("number", "size"))
+        self.assert_eq(pser.swaplevel("color", "size"), kser.swaplevel("color", "size"))
+
+        # Error conditions
+        self.assertRaises(IndexError, lambda: ks.Series([1, 2]).swaplevel())
```
Comment (ueshin) [inline databricks/koalas/tests/test_series.py:1854]: This should raise an `AssertionError`:

```py
>>> pd.Series([1, 2]).swaplevel()
Traceback (most recent call last):
...
AssertionError
```

File: databricks/koalas/series.py [hunk 7]
```diff
@@ -2440,6 +2440,76 @@ def sort_index(
...
+        1       a       x
+        2       b       y
+        dtype: object
+        >>> kser.swaplevel('number', 'word')
+        number  word
+        1       a       x
+        2       b       y
+        dtype: object
+        """
+        assert copy
```
Comment (ueshin) [inline databricks/koalas/series.py:2489]: Shall we explicitly do `assert copy is True`?

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

File: databricks/koalas/tests/test_series.py (hunk 6 above)
Comment (xinrong-databricks) [reply databricks/koalas/tests/test_series.py]: Makes sense! Modified.

File: databricks/koalas/series.py (hunk 7 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: Sounds good.

File: databricks/koalas/series.py (hunk 5 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: Thanks for explaining!

I'm a little confused by "contains". Their `spark_frame` seems to be identical.
```
>>> kdf.b._internal.spark_frame.show()
+-----------------+-----------------+---+---+-----------------+
|__index_level_0__|__index_level_1__|  a|  b|__natural_order__|
+-----------------+-----------------+---+---+-----------------+
|                a|                1|  1|  x|      42949672960|
|                b|                2|  2|  y|      94489280512|
+-----------------+-----------------+---+---+-----------------+

>>> kdf.b._kdf._internal.spark_frame.show()
+-----------------+-----------------+---+---+-----------------+
|__index_level_0__|__index_level_1__|  a|  b|__natural_order__|
+-----------------+-----------------+---+---+-----------------+
|                a|                1|  1|  x|      42949672960|
|                b|                2|  2|  y|      94489280512|
+-----------------+-----------------+---+---+-----------------+

```

Comment (ueshin) [reply databricks/koalas/series.py]: I mean their metadata:

```py
>>> kdf.b._internal.column_labels
[('b',)]
>>> kdf.b._kdf._internal.column_labels
[('a',), ('b',)]
```

or 

```py
>>> kdf.b._internal.data_spark_columns
[Column<b'b'>]
>>> kdf.b._kdf._internal.data_spark_columns
[Column<b'a'>, Column<b'b'>]
```

Comment (xinrong-databricks) [reply databricks/koalas/series.py]: I see, that's clear to me now. Thank you!

**Pull request merged**